src/chain.h

```diff
@@ -307,6 +308,17 @@ class CBlockIndex
     //! Will be std::nullopt if and only if nChainTx is zero.
```

Comment:
I don't understand why we made this optional. The Orchard chain value pool balance of an empty block chain is perfectly well-defined: it's zero. And similarly for `nChainLockboxValue`.

Reply:
I believe that it was because we couldn't rely on them being present when loading the block index from disk; we in all cases wanted to make sure that there was no chance that these values could be initialized to a default value from a non-default state.

Reply:
They are optional for two reasons:

- We do not store them on disk, and thus do not know their value when we load individual block data from disk. Once LoadBlockIndex is complete, we have them set.

- If we receive a block before its parent, we cannot compute this value, so it must be set to None untill we receive its parent and can process that parent's descendents (as we do in ReceivedBlockTransactions).

Reply:
Yes I think so.